WebAuthn WD-05 API updates

apowers313 opened the pull request:
Updates to tests based on the WD-05 spec. These are mostly the tests that were run at the interop, with a few new ones added: secure context tests, `credentials.get()` tests, and passing bad `rp` arguments to `credentials.create()`.

<!-- Reviewable:start -->

<!-- Reviewable:end -->

apowers313 committed:
changes from WD-05 informal interop

apowers313 committed:
basic secure context checking

apowers313 committed:
simple passing tests

apowers313 committed:
removing old tests

apowers313 committed:
removing old tests

apowers313 committed:
.get() validation and general clean-up

apowers313 committed:
initial testing of rp arguments

w3c-bots commented:
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/18885)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/280271835)

apowers313 committed:
removing stray console.log to keep w3c delinter happy

jcjones committed:
Updates per WD-07.

Note in Firefox Nightly these changes cause both getcredential-passing.https.html
and getcredential-passing.https.html to pass, and most of interfaces.https.html
to pass as well.

createcredential-badargs-rp.https.html has some other issues which should be
addressed in a different commit.

jcjones commented:
Hi, I'm one of the WebAuthn editors, and implementing this feature in Firefox.

This patch is good, though we also need to make some updates to come to WD-07. I submitted a different PR to @apowers313 at https://github.com/apowers313/web-platform-tests/pull/1 to add one more patch on top of these, but I'm fine with merging this and then opening a PR to pull in b6807f1a3e925faed8428dd8fd1aa20a027f6a03

apowers313 committed:
Merge pull request #1 from jcjones/webauthn-wd07-api-updates

Updates per WD-07.

jcjones reviewed (approved):
All of these need more work, but they are strictly better than what is currently in-tree, so let's move this forward.
